control-plane/agents/src/bin/core/pool/specs.rs

```diff
@@ -149,6 +149,8 @@ impl SpecOperationsHelper for ReplicaSpec {
             }
             ReplicaOperation::Unshare => Ok(()),
             ReplicaOperation::OwnerUpdate(_) => Ok(()),
+            // XXX: Shall we do something here with input size?
+            ReplicaOperation::Resize(_) => Ok(()),
```

Comment:
Yeah, probably can add a validation here itself?

Reply:
> Yeah, probably can add a validation here itself?

Added on the VolumeOperation check step.

Reply:
While the `ReplicaStatus` isn't `Online`, there could be a possibility that underlying lvol is healthy/reachable and hence resizable? so perhaps let's not block the flow at this point ?

Reply:
hmm should be very small possibility that the lvol is healthy if the status is not online. This should actually be checked in the scheduling, the replicas should be Online.
Also we should also check if the node is cordoned?

Reply:
> hmm should be very small possibility that the lvol is healthy if the status is not online. This should actually be checked in the scheduling, the replicas should be Online. Also we should also check if the node is cordoned?

@tiagolobocastro Not sure on cordon check. Cordon is to prevent scheduling new resources on a node right? So if a replica is already existing on that node we should let it resize I think.

Reply:
Yeah, makes sense